atom/common/api/resources/windows_bindings.js

```diff
@@ -46,6 +47,57 @@ var binding = {
     ipc.send('chrome-windows-update', responseId, windowId, updateInfo)
   },
 
+  get: function (windowId, getInfo, cb) {
+    console.warn('chrome.windows.get is not supported yet')
+  },
+
+  getLastFocused: function (getInfo, cb) {
+    console.warn('chrome.windows.getLastFocused is not supported yet')
+  },
+
+  remove: function (windowId, cb) {
+    console.warn('chrome.windows.remove is not supported yet')
+  },
+
+  onCreated: {
+
+    addListener: function (callback) {
+      console.warn('chrome.windows.onCreated is not supported yet')
+    },
+
+    Filters: ['app', 'normal', 'panel', 'popup']
+  },
+
+  onRemoved: {
+
+    addListener: function (callback) {
+      console.warn('chrome.windows.onRemoved is not supported yet')
+    },
+
+    Filters: ['app', 'normal', 'panel', 'popup']
+  },
+
+  onFocusChanged: {
+
+    addListener: function (cb) {
+
+      if(cb.length != 1) {
+        console.warn('Callback has wrong number of arguments.')
+      }
+
+      var responseId = ++focusId
+
+      ipc.on('chrome-windows-focused-window-update-response-' + responseId, function (evt, responseId) {
+        cb(responseId);
+      })
+
+      ipc.send('chrome-windows-focused-window-update', responseId)
+
+    },
+
+    Filters: ['app', 'normal', 'panel', 'popup']
```

Comment:
This was one of the issues I wanted to discuss. As far as I can tell, the Chrome API is in error here: it looks like they copy-pasted the text from onRemoved without updating it. While this text makes sense in the context of onRemoved, it doesn't shed any light on how onFocusChanged should work in the context of filters.

At any rate, it isn't clear how window filters should work for onFocusChanged, if at all. The basic definitions I can think of to propose don't make sense if you follow them all the way through. Should this property even be present at all? Does someone know how it should work?